### DISCUSSION TITLE
Problem with deploying sites on custom server

### DISCUSSION THREAD

**Discussion by mahemmati** [vitodeploy/vito#59] [Q&A]
Hi there .. 
I've installed Vito on a fresh Ubuntu on my own VPS then tried to Deploy a Laravel application to another fresh Ubuntu on another own VPS ... Vito provisions second server successfully and all services are up and running;
![image](https://github.com/vitodeploy/vito/assets/128333525/9bf07031-b734-446d-ba78-c814eb60b633)
As I deploy my Laravel application to the server ... following problems occur:

1. All deployments log contain the below error: (tried several deployment scripts and .env and nearly sure there is no issue with them)
![image](https://github.com/vitodeploy/vito/assets/128333525/bed4f76b-a243-4c77-bca3-b7d14b112914)
2. Vito does not composer install and I should do it manually through SSH (might be due to the prompt that composer rises when composer install with root user)
![image](https://github.com/vitodeploy/vito/assets/128333525/f49ecbda-0543-45e9-8a4c-d0fee691c4dd)
4. Vito does not pull from repo after the initial deploy and It is not doable via SSH either (tried to insert git pull on deployment script but did not work out)
On cli:
![image](https://github.com/vitodeploy/vito/assets/128333525/c50dd3e0-1420-4c5c-baa7-648cfe118066)
Also tried `git config --global --add safe.directory /home/vito/*********.com` but did not work out ...!

Any ideas?

**Comment by saeedvaziry** [comment]
Hey,

Could you share your deployment script?

Vito does run the composer install when creating new site, You just need to check the `composer install` checkbox.

from the screenshots, I see that you've ran some commands with root user under `vito` folder which most probably messed up with file permissions over there and the owner of everything needs to be set back to `vito` user

Also Vito clones the website on the website installation step and if it fails the site cannot be installed at all.

**Reply by mahemmati** [reply]
1. The deployment script is as follow:

composer install --no-dev
php artisan down
php artisan migrate --force
php artisan config:clear
php artisan cache:clear
php artisan config:cache
php artisan up

2. About running commands as `root` .. yes I did. As in adding the public key to server, `vito` requires root user priv. I assumed the others also require it. I'll retry as nonroot and feed back soon.
3. Cloning works ... but `git pull`s don't.

**Reply by saeedvaziry** [reply]
Could you also let me know which Source Control provider you're using? So I  can try to reproduce

**Reply by mahemmati** [reply]
It is the GitHub ... 
I tried several scenarios with different deployment scripts. Each time it pulls correctly on the first deploy but fails on the consecutive tries.
![image](https://github.com/vitodeploy/vito/assets/128333525/2685df20-92ef-4d5a-81d2-abe3ff6cc1e5)

**Reply by saeedvaziry** [reply]
Please raise an issue then

**Comment by saeedvaziry** [comment]
The issue has been fixed in #61 

You need to update VitoDeploy

https://vitodeploy.com/introduction/update.html
